internal/controllers/addon/phase_ensure_secretpropagation.go

```diff
@@ -21,15 +22,15 @@ func (r *AddonReconciler) ensureSecretPropagation(
 	ctx context.Context, log logr.Logger,
 	addon *addonsv1alpha1.Addon,
 ) (requeueResult, error) {
-	if addon.Spec.SecretPropagation == nil {
-		// nothing to do
-		return resultNil, nil
+	var secrets []addonsv1alpha1.AddonSecretPropagationReference
+	if addon.Spec.SecretPropagation != nil {
+		secrets = addon.Spec.SecretPropagation.Secrets
```

Comment:
Can we just wrap the secret clean-up logic in a separate function, and call it in a defer block here? That way, the remaining secret propagation logic is never executed, while secrets are also always cleaned up (we could revert back to the original `addon.Spec.SecretPropagation == nil` check).

Does it make more sense to do it this way? WDYT?